src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -171,150 +181,206 @@ export function DiscoverLegacy({
+                    data={opts.data}
+                    error={fetchError}
                   />
-                  {opts.timefield && (
-                    <TimechartHeader
-                      dateFormat={opts.config.get('dateFormat')}
-                      timeRange={timeRange}
-                      options={search.aggs.intervalOptions}
-                      onChangeInterval={onChangeInterval}
-                      stateInterval={state.interval || ''}
-                      bucketInterval={bucketInterval}
-                    />
-                  )}
-
-                  {opts.timefield && (
-                    <section
-                      aria-label={i18n.translate('discover.histogramOfFoundDocumentsAriaLabel', {
-                        defaultMessage: 'Histogram of found documents',
-                      })}
-                      className="dscTimechart"
-                    >
-                      {opts.chartAggConfigs && rows.length !== 0 && (
-                        <div className="dscHistogram" data-test-subj="discoverChart">
-                          <DiscoverHistogram
-                            chartData={histogramData}
-                            timefilterUpdateHandler={timefilterUpdateHandler}
-                          />
-                        </div>
-                      )}
-                    </section>
-                  )}
-
-                  <div className="dscResults">
-                    <section
-                      className="dscTable dscTableFixedScroll"
-                      aria-labelledby="documentsAriaLabel"
-                      ref={fixedScrollRef}
-                    >
-                      <h2 className="euiScreenReaderOnly" id="documentsAriaLabel">
-                        <FormattedMessage
-                          id="discover.documentsAriaLabel"
-                          defaultMessage="Documents"
-                        />
-                      </h2>
-                      {rows && rows.length && (
-                        <div className="dscDiscover">
-                          <DocTableLegacy
-                            columns={state.columns || []}
-                            indexPattern={indexPattern}
-                            minimumVisibleRows={minimumVisibleRows}
-                            rows={rows}
-                            sort={state.sort || []}
-                            searchDescription={opts.savedSearch.description}
-                            searchTitle={opts.savedSearch.lastSavedTitle}
-                            onAddColumn={addColumn}
-                            onFilter={onAddFilter}
-                            onMoveColumn={onMoveColumn}
-                            onRemoveColumn={onRemoveColumn}
-                            onSort={onSort}
+                )}
+                {resultState === 'uninitialized' && <DiscoverUninitialized onRefresh={fetch} />}
```

Comment:
Can we use ternary statements here, I think it would improve readability?
`resultState === 'uninitialized' ?  <DiscoverUninitialized onRefresh={fetch} : null`

Reply:
But I think then we should do it the same way throughout the file? for consistency. the readability is subjective, but what's an argument for ternary is, that it's explicit. issue with `&&` are very well described here: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx 
However I'd prefer changes here later on (since it's not a single line)